MAUI/DataGrid/Column-types.md

```diff
@@ -0,0 +1,916 @@
+{% endtabs %}
+
+The following code example illustrates how template column can be used to load a stock cell inside it:
+
+{% highlight xaml %}
+<ContentPage.Resources>
+    <ResourceDictionary>
+        <local:ImageConverter x:Key="imageConverter" /> 
+    </ResourceDictionary>
+</ContentPage.Resources>
+
+<ContentPage.ContentView>
+    <syncfusion:SfDataGrid x:Name="dataGrid"
+                           ColumnSizer="Star">
+
+        <syncfusion:SfDataGrid.Columns>
+            <syncfusion:DataGridTemplateColumn="Stock Change"
+                                           MappingName="StockChange">
+                <syncfusion:DataGridTemplateColumn.CellTemplate>
+                    <DataTemplate>
+                        <Grid>
+                            <Grid.ColumnDefinitions>
+                                <ColumnDefinition Width="Auto" />
+                                <ColumnDefinition Width="*" />
+                            </Grid.ColumnDefinitions>
+                            <Image Grid.Column="0"
+                                   Source="{Binding StockChange,
+                                   Converter={StaticResource imageConverter}}" />
+                            <Label x:Name="changeValue" Grid.Column="1"
+                                   Text="{Binding StockChange}" TextColor="Black"
+                                   XAlign="Center" YAlign="Center">
+                            </Label>
+                        </Grid>
+                    </DataTemplate>
+                </syncfusion:DataGridTemplateColumn.CellTemplate>
+            </syncfusion:DataGridTemplateColumn>
+        </syncfusion:SfDataGrid.Columns>
+    </syncfusion:SfDataGrid>
+</ContentPage.ContentView>
+{% endhighlight %}
+
+In order to get the above code example working, write a converter to load images inside the `GridCell` based on the CellValue. The images that have to be loaded in the `GridCell` must be added as EmbeddedResource.
+
+The converter code for loading images in a template column is shown in the following code:
+
+{% highlight c# %}
+public class ImageConverter:IValueConverter
+{
+    public object Convert (object value, Type targetType, object parameter, CultureInfo culture)
+    {
+        var data = value as double?;
+        if (data != null && data > 0)
+            return ImageSource.FromResource("DataGridSample.Icons.Green.png");
+        else
+            return ImageSource.FromResource("DataGridSample.Icons.Red.png");
+    }
+
+    public object ConvertBack (object value, Type targetType, object parameter, CultureInfo culture)
+    {
+        throw new NotImplementedException ();
+    }
+} 
+{% endhighlight %}
+
+The following screenshot shows the different types of columns in the SfDataGrid:
+
+### CellTemplateSelector
+
+Load the underlying records through the `CellTemplate` to use CellTemplateSelector. The following code example shows templating of the DataGridTemplateColumn using the `CellTemplate` property:
```

Comment:
Load the underlying records through the `CellTemplate` to use the CellTemplateSelector. The following code example shows templating of the DataGridTemplateColumn using the `CellTemplate` property: